civiform/civiform
# [Program Card Images] Add breadcrumbs to admin program image upload page

### Description

This PR makes use of the [USDWS breadcrumb component](https://designsystem.digital.gov/components/breadcrumb/) to add breadcrumbs to the admin program image upload page. The breadcrumbs are "Edit programs > {Program Name} > Program Image Upload". It defines a generic `BreadcrumbFactory` class that can create breadcrumbs for any content. Right now, `ProgramImageView` is the only class using `BreadcrumbFactory` but I expect `BreadcrumbFactory` will be used by other pages in the future.

[Here is the Figma](https://www.figma.com/file/rsQEvXUCuTQwyECapaKzw9/CiviForm---Image-upload-to-program-cards?type=design&node-id=179-6429&mode=design&t=Ug9DE2f0zwgPoX5E-0) with the breadcrumbs design.

## Release notes

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)


#### User visible changes


#### New Features

### Instructions for manual testing

1. Enable the PROGRAM_CARD_IMAGES flag.
2. As an admin, edit a program then click the "Edit program image" button.
3. Verify the program image page has breadcrumbs (as shown in the screenshots).

### Issue(s) this completes

Epic #5676 

Sub-task #5999
